controllers/url_controller.js

```diff
@@ -0,0 +1,62 @@
+import Url from "../models/Url.js";
+
+const shorten_url = async (req, res, next) => {
+    try{
+        let { original_url } = req.body;
+
+        if (!original_url) {
+            return res.status(400).send({ error: 'A URL is required.' });
+        }
+
+        // Remove all whitespace from the URL and trim it
+        original_url = original_url.replace(/\s+/g, '').trim();
```

Comment:
The .trim() function could be seemed as redundant here because the .replace() function already removes all whitespace from the original_url string.